internal/features/stacks/ast/deploy.go

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package ast
+
+import (
+	"strings"
+
+	"github.com/hashicorp/hcl/v2"
+	globalAst "github.com/hashicorp/terraform-ls/internal/terraform/ast"
+)
+
+type DeployFilename string
+type DeployFiles map[DeployFilename]*hcl.File
+type DeployDiags map[DeployFilename]hcl.Diagnostics
+type SourceDeployDiags map[globalAst.DiagnosticSource]DeployDiags
+
+func (df DeployFilename) String() string {
+	return string(df)
+}
+
+func (df DeployFilename) IsJSON() bool {
+	return strings.HasSuffix(string(df), ".json")
+}
+
+func (df DeployFilename) IsIgnored() bool {
+	return globalAst.IsIgnoredFile(string(df))
+}
+
+func IsDeployFilename(name string) bool {
+	return strings.HasSuffix(name, ".tfdeploy.hcl") ||
+		strings.HasSuffix(name, ".tfdeploy.json")
+}
+
+func (df DeployFiles) Copy() DeployFiles {
+	m := make(DeployFiles, len(df))
+	for name, file := range df {
+		m[name] = file
+	}
+	return m
+}
+
+func (dd DeployDiags) Copy() DeployDiags {
+	m := make(DeployDiags, len(dd))
+	for name, diags := range dd {
+		m[name] = diags
+	}
+	return m
+}
+
+func (dd DeployDiags) AutoloadedOnly() DeployDiags {
```

Comment:
What does Autoloaded mean? I'm sure this concept existed before, but I could really use a half sentence comment here that explains what autoloaded means

Reply:
Added a commit with a half sentence documenting these